updater/internal/install/install.go

```diff
@@ -0,0 +1,163 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package install
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+// Installer allows you to install files from latestDir into installDir,
+// as well as update the service configuration using the "Install" method.
+type Installer struct {
+	latestDir  string
+	installDir string
+	svc        Service
+}
+
+// NewInstaller returns a new instance of an Installer.
+func NewInstaller(tempDir string) (*Installer, error) {
+	latestDir := filepath.Join(tempDir, "latest")
+	installDirPath, err := installDir()
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine install dir: %w", err)
+	}
+
+	return &Installer{
+		latestDir:  latestDir,
+		svc:        newService(latestDir),
+		installDir: installDirPath,
+	}, nil
+}
+
+// Install installs the unpacked artifacts in latestDirPath to installDirPath,
+// as well as installing the new service file using the provided Service interface
+func (i Installer) Install() error {
+	// Stop service
+	if err := i.svc.Stop(); err != nil {
+		return fmt.Errorf("failed to stop service: %w", err)
+	}
+
+	// install files that go to installDirPath to their correct location,
+	// excluding any config files (logging.yaml, config.yaml, manager.yaml)
+	if err := moveFiles(i.latestDir, i.installDir); err != nil {
+		return fmt.Errorf("failed to install new files: %w", err)
+	}
+
+	// Uninstall previous service
+	if err := i.svc.Uninstall(); err != nil {
+		return fmt.Errorf("failed to uninstall service: %w", err)
+	}
+
+	// Install new service
+	if err := i.svc.Install(); err != nil {
+		return fmt.Errorf("failed to install service: %w", err)
+	}
+
+	// Start service
+	if err := i.svc.Start(); err != nil {
+		return fmt.Errorf("failed to start service: %w", err)
+	}
+
+	return nil
+}
+
+func moveFiles(latestDirPath, installDirPath string) error {
+	err := filepath.WalkDir(latestDirPath, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+
+		if skipFile(path) {
+			return nil
+		}
```

Comment:
nit: this might read better as a switch statement

```go
switch {
  case err != nil:
      return err
  case d.IsDir():
      return nil
  case skipFile(path):
      return nil
}
```